### PR TITLE
OAuth: Add custom unauthorized message option in configuration

### LINKED ISSUE #84639: [Feature Request] Improve error message when an unauthorized user attempts to log in to Grafana instance
Coming from VOC: see [slide](https://docs.google.com/presentation/d/1Rr3LFBk24Twy0SQi_lNY4-iBQopWN7xWzTUglBzCUIw/edit#slide=id.g2667a96c7fe_0_367)

From [Josh](https://raintank-corp.slack.com/archives/C025WTVRBSN/p1713864835505749?thread_ts=1713861768.035349&cid=C025WTVRBSN):
* Much of the issues reported in the VOC issue have been addressed over time
* The oauth "login provider denied login request" error message is kinda wonky, and should be improved
* The mechanism that specifically oauth error messages get to the frontend is different from others, because it's set in a cookie
* I think the backend should be refactored to use errutil.Unauthorized("error.code", errutil.WithPublicMessage("....")) , and set an error code which we can translate on the frontend
* I think this is relatively straight forward, but might require some back and forth with Identity. Maybe we could get thumbs up for approach, and we implement it

### PR DESCRIPTION
**Pull request by tskarhed**
**What is this feature?**

- Adds translations for the default error message
- Adds a config option for a custom error message

**Why do we need this feature?**

A common issue is that the user isn't set up properly with the oAuth system, and needs someone to do something about it. Setting a custom message may help informing about where to reach out for support.

**Who is this feature for?**

Admins who want to inform there users where to go, reducing the support workload

**Which issue(s) does this PR fix?**:

Fixes [#84639](https://github.com/grafana/grafana/issues/84639)

**Special notes for your reviewer:**

The easiest way to review this, is to move the line of code returning the error to before the if-statement and try to login using oAuth an oAuth provider (you don't actually have to set one up if you do this)

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/contribute/release-notes/#how-to-determine-if-content-belongs-in-whats-new), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/contribute/release-notes/) doc.

### CODE REVIEW

File: docs/sources/setup-grafana/configure-grafana/_index.md [hunk 1]
```diff
@@ -964,6 +964,10 @@ This setting is ignored if multiple OAuth providers are configured. Default is `
 How many seconds the OAuth state cookie lives before being deleted. Default is `600` (seconds)
 Administrators can increase this if they experience OAuth login state mismatch errors.
 
+### oauth_login_error_message
+
+A custom error message for when users are unauthorized. Default is a key for an internationalized phrase in the frontend, `Login provider denied login request`.
```
Comment (tskarhed) [inline docs/sources/setup-grafana/configure-grafana/_index.md:969]: Is this just a technicality? Technically the default is "oauth.login.error", which is then set as `Login provider denied login request` or a translated version.

File: conf/defaults.ini [hunk 2]
```diff
@@ -582,6 +582,9 @@ oauth_auto_login = false
 # OAuth state max age cookie duration in seconds. Defaults to 600 seconds.
 oauth_state_cookie_max_age = 600
 
+# Sets a custom oAuth error message. This is useful if you need to point the users to a specific location for support.
+oauth_login_error_message =
```
Comment (kalleep) [inline conf/defaults.ini:586]: Even though we default to `oauth.login.error` when reading the config we should also set it here

File: conf/sample.ini [hunk 3]
```diff
@@ -583,6 +583,9 @@
 # Deprecated, use auto_login option for specific provider instead.
 ;oauth_auto_login = false
 
+# Sets a custom oAuth error message. This is useful if you need to point the users to a specific location for support.
+;oauth_login_error_message =
```
Comment (kalleep) [inline conf/sample.ini:587]: Same here for consistency

File: docs/sources/setup-grafana/configure-grafana/_index.md (hunk 1 above)
Comment (DanCech) [reply docs/sources/setup-grafana/configure-grafana/_index.md:969]: This is fine imho, the internal string isn't something an end user really needs to know or care about.

**Review by DanCech — APPROVED** [review]

**Review by kalleep — APPROVED** [review]

**Review by ashharrison90 — APPROVED** [review]
lgtm 👍

**Pull request merged**